src/main/java/uk/gov/companieshouse/api/accounts/validation/DebtorsValidator.java

```diff
@@ -0,0 +1,187 @@
+
+    private void validateRequiredCurrentPeriodTotalFieldNotNull(Debtors debtors, Errors errors) {
+
+        if ((debtors.getCurrentPeriod().getTradeDebtors() != null ||
+            debtors.getCurrentPeriod().getPrepaymentsAndAccruedIncome() != null ||
+            debtors.getCurrentPeriod().getOtherDebtors() != null ||
+            debtors.getCurrentPeriod().getGreaterThanOneYear() != null ||
+            !debtors.getCurrentPeriod().getDetails().isEmpty()) &&
+            debtors.getCurrentPeriod().getTotal() == null) {
+
+            addError(errors, invalidNote, CURRENT_TOTAL_PATH);
+        }
+    }
+
+
+    private void validateRequiredPreviousPeriodTotalFieldNotNull(Debtors debtors, Errors errors) {
+
+        if ((debtors.getPreviousPeriod().getTradeDebtors() != null ||
+            debtors.getPreviousPeriod().getPrepaymentsAndAccruedIncome() != null ||
+            debtors.getPreviousPeriod().getOtherDebtors() != null ||
+            debtors.getPreviousPeriod().getGreaterThanOneYear() != null) &&
+            debtors.getPreviousPeriod().getTotal() == null) {
+
+            addError(errors, invalidNote, PREVIOUS_TOTAL_PATH);
+        }
+    }
+
+    private void validateCurrentPeriodTotalIsCorrect(Debtors debtors, Errors errors) {
+
+        if (debtors.getCurrentPeriod().getTotal() != null) {
+            Long traderDebtors = Optional.ofNullable(debtors.getCurrentPeriod().getTradeDebtors()).orElse(0L);
+            Long prepayments = Optional.ofNullable(debtors.getCurrentPeriod().getPrepaymentsAndAccruedIncome()).orElse(0L);
+            Long otherDebtors = Optional.ofNullable(debtors.getCurrentPeriod().getOtherDebtors()).orElse(0L);
+            Long moreThanOneYear = Optional.ofNullable(debtors.getCurrentPeriod().getGreaterThanOneYear()).orElse(0L);
+
+            Long total = debtors.getCurrentPeriod().getTotal();
+
+            Long sum = traderDebtors + prepayments + otherDebtors + moreThanOneYear;
+
+            validateAggregateTotal(total, sum, CURRENT_TOTAL_PATH, errors);
+        }
+    }
+
+    private void validatePreviousPeriodTotalIsCorrect(Debtors debtors, Errors errors) {
+
+        if (debtors.getPreviousPeriod().getTotal() != null) {
+            Long traderDebtors = Optional.ofNullable(debtors.getPreviousPeriod().getTradeDebtors()).orElse(0L);
+            Long prepayments = Optional.ofNullable(debtors.getPreviousPeriod().getPrepaymentsAndAccruedIncome()).orElse(0L);
+            Long otherDebtors = Optional.ofNullable(debtors.getPreviousPeriod().getOtherDebtors()).orElse(0L);
+            Long moreThanOneYear = Optional.ofNullable(debtors.getPreviousPeriod().getGreaterThanOneYear()).orElse(0L);
+
+            Long total = debtors.getPreviousPeriod().getTotal();
+
+            Long sum = traderDebtors + prepayments + otherDebtors + moreThanOneYear;
+
+            validateAggregateTotal(total, sum, PREVIOUS_TOTAL_PATH, errors);
+        }
+    }
+
+    public boolean isMultipleYearFiler(Transaction transaction) {
```

Comment:
Can this method be private or is the plan to use it elsewhere?

Reply:
Thinking about it - this method might be best placed in the BaseValidator - as future notes with multiple periods will need to validate this way also